otterai/indexer.py

```diff
@@ -42,9 +43,22 @@ def index_codebase(root_dir: str) -> Dict[str, List[str]]:
         'test': [],
         'other': []
     }
+
+    default_ignore_patterns = [
+        '*.pyc', '__pycache__/*', '.git/*', '.github/*', 'node_modules/*',
```

Comment:
🧹 The list of default ignore patterns contains duplicates (e.g., '*.pyc' and '__pycache__/*'). Remove duplicates for better maintainability.